src/meta/src/barrier/creating_job_control.rs

```diff
@@ -0,0 +1,450 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::mem::take;
+use std::sync::Arc;
+
+use itertools::Itertools;
+use risingwave_common::util::epoch::Epoch;
+use risingwave_pb::common::WorkerNode;
+use risingwave_pb::ddl_service::DdlProgress;
+use risingwave_pb::hummock::HummockVersionStats;
+use risingwave_pb::stream_service::BarrierCompleteResponse;
+use tracing::{debug, info, warn};
+
+use crate::barrier::command::CommandContext;
+use crate::barrier::info::InflightGraphInfo;
+use crate::barrier::progress::CreateMviewProgressTracker;
+use crate::barrier::rpc::ControlStreamManager;
+use crate::barrier::{
+    BarrierKind, Command, CreateStreamingJobCommandInfo, SnapshotBackfillInfo, TracedEpoch,
+};
+use crate::manager::WorkerId;
+use crate::model::ActorId;
+use crate::MetaResult;
+
+#[derive(Debug)]
+struct CreatingStreamingJobEpochState {
+    node_to_collect: HashSet<WorkerId>,
+    resps: Vec<BarrierCompleteResponse>,
+}
+
+#[derive(Debug)]
+pub(super) enum CreatingStreamingJobStatus {
+    ConsumingSnapshot {
+        prev_epoch_fake_physical_time: u64,
+        pending_commands: Vec<Arc<CommandContext>>,
+        version_stats: HummockVersionStats,
+        create_mview_tracker: CreateMviewProgressTracker,
+        graph_info: InflightGraphInfo,
+        /// The `prev_epoch` of pending non checkpoint barriers
+        pending_non_checkpoint_barriers: Vec<u64>,
+        snapshot_backfill_actors: HashMap<WorkerId, HashSet<ActorId>>,
+    },
+    ConsumingLogStore {
+        graph_info: InflightGraphInfo,
+    },
+    Finishing(u64), // The prev epoch that marks it as Finishing
+    Finished(u64),  // The prev epoch that marks it as Finished
+}
+
+#[derive(Debug)]
+pub(super) struct CreatingStreamingJobControl {
+    pub(super) info: CreateStreamingJobCommandInfo,
+    pub(super) snapshot_backfill_info: SnapshotBackfillInfo,
+    // key is prev_epoch of barrier
+    inflight_barrier_queue: BTreeMap<u64, CreatingStreamingJobEpochState>,
+    pub(super) collected_barrier: Vec<(u64, Vec<BarrierCompleteResponse>)>,
+    backfill_epoch: Epoch,
+    pub(super) status: CreatingStreamingJobStatus,
+}
+
+impl CreatingStreamingJobControl {
+    pub(super) fn new(
+        info: CreateStreamingJobCommandInfo,
+        snapshot_backfill_info: SnapshotBackfillInfo,
+        backfill_epoch: Epoch,
+        version_stat: &HummockVersionStats,
+    ) -> Self {
+        info!(
+            table_id = info.table_fragments.table_id().table_id,
+            definition = info.definition,
+            "new creating job"
+        );
+        let mut create_mview_tracker = CreateMviewProgressTracker::default();
```

Comment:
Yes, this is expected. Here we just want to reuse the logic of `CreateMviewProgressTracker ` so that we don't need to develop extra logic to track the backfill progress of snapshot backfill.